pkg/utils/kubeconfig/kubeconfig.go

```diff
@@ -213,6 +224,37 @@ type AWSAuthenticatorVersionFormat struct {
 	Version string `json:"Version"`
 }
 
+func awsCliIsAboveVersion() (bool, error) {
+	awsCliVersion := getAWSCLIVersion()
+	compareVersions, err := utils.CompareVersions(awsCliVersion, "2.0.0")
+	if err != nil {
+		return false, fmt.Errorf("failed to parse versions: %w", err)
+	}
+	if compareVersions < 0 {
+		compareVersions, err = utils.CompareVersions(awsCliVersion, AWSCLIv1MinimumBetaVersion)
+	} else {
+		compareVersions, err = utils.CompareVersions(awsCliVersion, AWSCLIv2MinimumBetaVersion)
+	}
+	if err != nil {
+		return false, fmt.Errorf("failed to parse versions: %w", err)
+	}
```

Comment:
It is most unfortunate that there are two separate versions of AWS cli that provide betav1. :D makes this code a tiny bit hard to parse. I had to do a double-take on this.

Reply:
Completely agree, added comment to the function. Thanks.